src/box/alter.cc

```diff
@@ -2789,6 +2793,9 @@ on_replace_dd_index(struct trigger * /* trigger */, void *event)
 					  "can not alter a referenced index");
 				return -1;
 			}
+			(void) txn_can_yield(txn, true);
```

Comment:
1. We need to remove txn_on_yield from the vinyl code (vinyl_build_index and vinyl_check_format).
2. I'd place this around space_build_index and space_check_format (probably added new wrappers). Rationale:
   - DDL is only allowed to yield in these functions so if we occasionally yield somewhere else, we'll get an error and hopefully a broken test.
   - By placing it at the high level, you might miss some places. E.g. TruncateIndex call space_build_index too and might yield, at least in Vinyl AFAIR.

Reply:
Ok, agree with both points. Updated the branch. However, see my comment below, seems still we may get dirty read..

Reply:
You forgot to remove the code enabling yields from on_replace_dd_{space,index}. Apparently, we don't need it anymore.

Also, I just checked - it turns out that memtx also enables yields in space_build_index / space_check_format so we need to remove these checks as well. Alternatively, we can leave them as is, both in memtx and vinyl, and remove it some time later, in a separate patch, probably along with txn_check_singlestatement. Then we wouldn't need to add any code enabling yields to alter.cc for now. Up to you.

Reply:
Sorry, cleaned up these places enabling yields from on_replace_dd triggers...

I'd prefer to follow current refactoring and remove code enabling yields from space/index internals and leave them only on top level at alter.cc. Branch is updated.